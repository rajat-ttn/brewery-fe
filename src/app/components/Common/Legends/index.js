import React from 'react';
import { PropTypes } from 'prop-types';

// Legends Component
const Legends = ({ legendsValue, legendColorClass }) => (
        <div className="colorInfoWrapper">
            <div className="colorInfo">
                {
                    legendsValue && legendsValue.map((legend, i) => (
                        <span key={i}>
                            <span className={`status-circle ${legendColorClass[i]}`} />
                            <small><strong>{legend}</strong></small>
                        </span>
                    ))
                }
            </div>
        </div>
);

Legends.propTypes = {
    legendsValue: PropTypes.array.isRequired,
    legendColorClass: PropTypes.array.isRequired,
};
export default Legends;