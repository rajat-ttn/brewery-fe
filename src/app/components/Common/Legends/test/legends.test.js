import React from 'react';
import { shallow } from 'enzyme';
import Legends from '../index';


describe('Legends Component should render correctly', () => {
    let allProps,
        legendComponent;

    beforeEach(() => {
        allProps = {
            legendsValue: ['Too Low'],
            legendColorClass: ['blueBackground'],
        };
        legendComponent = shallow(<Legends {...allProps}  />);
    });

    describe("render legend element", () => {
        it("always renders span", () => {
            const span = legendComponent.find("span").first();
            expect(span.length).toBe(allProps.legendsValue.length);
        });
    });

    it('renders correctly', () => {
        expect(legendComponent).toMatchSnapshot();
    });
});
