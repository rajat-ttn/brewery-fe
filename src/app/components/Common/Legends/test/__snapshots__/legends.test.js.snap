// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Legends Component should render correctly renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Legends
    legendColorClass={
        Array [
            "blueBackground",
          ]
    }
    legendsValue={
        Array [
            "Too Low",
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="colorInfo"
>
        <span>
                <span
                        className="status-circle blueBackground"
                />
                <small>
                        <strong>
                                Too Low
                        </strong>
                </small>
        </span>
</div>,
      "className": "colorInfoWrapper",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span>
            <span
                        className="status-circle blueBackground"
            />
            <small>
                        <strong>
                                    Too Low
                        </strong>
            </small>
</span>,
        ],
        "className": "colorInfo",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="status-circle blueBackground"
/>,
              <small>
                <strong>
                                Too Low
                </strong>
</small>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "status-circle blueBackground",
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <strong>
                  Too Low
</strong>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Too Low",
                },
                "ref": null,
                "rendered": "Too Low",
                "type": "strong",
              },
              "type": "small",
            },
          ],
          "type": "span",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="colorInfo"
>
          <span>
                    <span
                              className="status-circle blueBackground"
                    />
                    <small>
                              <strong>
                                        Too Low
                              </strong>
                    </small>
          </span>
</div>,
        "className": "colorInfoWrapper",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span>
              <span
                            className="status-circle blueBackground"
              />
              <small>
                            <strong>
                                          Too Low
                            </strong>
              </small>
</span>,
          ],
          "className": "colorInfo",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="status-circle blueBackground"
/>,
                <small>
                  <strong>
                                    Too Low
                  </strong>
</small>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "status-circle blueBackground",
                },
                "ref": null,
                "rendered": null,
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <strong>
                    Too Low
</strong>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Too Low",
                  },
                  "ref": null,
                  "rendered": "Too Low",
                  "type": "strong",
                },
                "type": "small",
              },
            ],
            "type": "span",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
